test/utils.go

```diff
@@ -35,17 +35,17 @@ func createTempDir(t *testing.T) (string, string, func()) {
 	return daemonPath, clientPath, closer
 }
 
-func createDaemon(t *testing.T, daemonPath string) (*p2pd.Daemon, func()) {
+func createDaemon(t *testing.T, daemonAddr ma.Multiaddr) (*p2pd.Daemon, func()) {
```

Comment:
these `create*` functions can just call `t.Fatal` instead of returning an error

Reply:
Updated